src/Oscoin/Consensus/Nakamoto.hs

```diff
@@ -192,7 +192,7 @@ chainDifficulty (NonEmpty.fromList -> blks)
     | NonEmpty.length blks `mod` fromIntegral difficultyBlocks == 0 =
         encodeDifficulty $ fst (decodeDifficulty currentDifficulty)
                          * fromIntegral targetElapsed
-                         `div` toInteger actualElapsed
+                         `div` (max 1 (toInteger actualElapsed)) -- avoids division by 0.
```

Comment:
Strangely enough, I had to do this to avoid a division by 0 that CI was consistently reporting. I have no clue why we didn't catch this earlier, if this is even possible and perhaps it's just a misbehaviour of the fact #421 is still pending or what 🤔 

_me wishes we had Liquid Haskell._

Reply:
Uh - that's strange

Reply:
I think the correct fix is to not change the difficulty unless the prefix length is > `blocksConsidered`

Reply:
It might be worthwhile giving this another shot now that we have merged #421, perhaps that was the culprit 🤔

Reply:
Also as mentioned on IRC, we should be checking that at least 2016 blocks exist before changing the difficulty (separate issue but may be related).